addons/hr_expense/tests/test_expenses.py

```diff
@@ -894,59 +894,6 @@ def test_employee_supplier(self):
             'partner_id': self.expense_user_employee.partner_id.id,
         }])
 
-    def test_print_expense_check(self):
-        """
-        Test the check content when printing a check
-        that comes from an expense
-        """
```

Comment:
Why is this test removed? It seems unrelated to your commit?
And your new test doesn't seem to do the same kind of assertions

Reply:
Because it consider `account_check_printing` is installed, which is the case on runbot but hr_expense doesn't depends on it

Reply:
Just do something like this then
```python
if 'check_number' not in self.env['account.payment']._fields:
    self.skipTest('`account_check_printing` is not installed')
```

Reply:
QDP asked me to remove that change from my refactor (I was directly checking the state of the module though)
And because the payment is what handles that part, another test is taking care of the payment values so no need to check further for these imo
And the new test is handling the rest (when it's green again)

Reply:
> QDP asked me to remove that change from my refactor

Why?

> another test is taking care of the payment values

Which?

Reply:
- https://github.com/odoo/odoo/pull/130244#discussion_r1365408382
- I admit I thought it was more explicit than that but the move from the payment is tested here (the tests are trash since forever, hence their rework in 17.0) https://github.com/odoo/odoo/blob/d8ea48fbefbc8fb1528581d3c3d939f89b7feab9/addons/hr_expense/tests/test_expenses.py#L519-L646

Reply:
It is really not obvious to me where you check that we have a reference to the move from which the payment has been created in that test.
Also, the fact that it has been removed in 17.0 (which I still think is a mistake) doesn't mean it should be removed here?

Reply:
> There is no special behaviour between expense and account_check_printing so I really don't get why we need that test to begin with.

Obviously there is something that didn't work: https://www.odoo.com/web#id=3044141&model=project.task
Something doesn't work: not trivial and might break -> test.
Now of course the test was not optimal as it should have been written in `account_check_printing` by creating a payment for a receipt.
But we don't just remove tests like that, especially when it is completely unrelated. The test still runs when installing only `hr_expense` because the main company is in US and so it installs `account_check_printing` by default.

So either don't touch this or move it where it belongs. But don't delete.

Reply:
Re-added the test in a slightly edited version